holiday-management/src/services/db-service.ts

```diff
@@ -141,12 +141,41 @@ export class DbService {
                     const db = client.db(appConfig.mainDbName);
                     const collection = db.collection(holidayRequestCollection);
                     let request = userId === undefined ? {} : { 'userId': userId, 'isActive': true };
+                    let sort = { creationDate: -1 };
+
+                    collection.find(request).sort(sort).toArray((err, result) => {
+                        client.close();
 
-                    collection.find(request).toArray((err, result) => {
                         if (err) { throw err; }
+                        resolve(result);
+                    });
+                })
+                .catch(error => reject(error));
+        });
+    }
 
+    getHolidayRequest(id: string): Promise<HolidayRequest> {
+        return new Promise<HolidayRequest>((resolve, reject) => {
```

Comment:
I think this wrapping promise is redundant. We can return promise chain started by `MongoClient.connect` and `return document;` on line 177 instead resolve. Catch on line 152 is then not needed

Reply:
Simplified